src/lib.rs

```diff
@@ -124,6 +124,13 @@ impl XorName {
         XOR_NAME_BITS
     }
 
+    /// Returns a copy of `self`, with the `i`-th bit flipped.
+    pub fn with_flipped_bit(&self, index: usize) -> XorName {
+        let &XorName(mut bytes) = self;
+        bytes[index / 8] = bytes[index / 8] ^ (1 << (7 - index % 8));
```

Comment:
I don't think so: An index outside that range will always be a programming error, so I'd vote for panic.
But I'll definitely add that to the method's documentation.